app/code/Magento/Bundle/Model/Product/CopyConstructor/Bundle.php

```diff
@@ -26,8 +26,16 @@ public function build(Product $product, Product $duplicate)
 
         $bundleOptions = $product->getExtensionAttributes()->getBundleProductOptions() ?: [];
         $duplicatedBundleOptions = [];
+        
+        /**
+         * @var string $key
+         * @var \Magento\Bundle\Model\Option $bundleOption
+         */
         foreach ($bundleOptions as $key => $bundleOption) {
-            $duplicatedBundleOptions[$key] = clone $bundleOption;
+            $duplicatedBundleOption = clone $bundleOption;
+            $duplicatedBundleOption->unsetData('option_id');
+
+            $duplicatedBundleOptions[$key] = $duplicatedBundleOption;
```

Comment:
1) `Error: Call to undefined method Mock_BundleOptionInterface_3055e245::unsetData()` only methods from interface should be used, probably `setOptionId(null)` or something can do the trick
2) Please specify typehint for `$bundleOptions` instead (`\Magento\Bundle\Model\Option[]`)
3) Codacy is complaining about long names, probably `clonedBundleOptions` will work
4) Adopt tests to changed code, write a separate test method to enforce `option_id` is unset
5) Use commit amend + force push so that after all changes there is only one commit